_posts/2017-08-08-euroclojure-2017.md

```diff
@@ -0,0 +1,209 @@
+
+![Testing Operational Transformation](/img/euroclojure2017/testing-ot.png)
+
+### Clojure Tools for Symbolic Artificial Intelligence, Saul Johnson
+
+Symbolic artificial intelligence uses a human readable representation for problems and logic. Symbolic pattern matching, operator search and legal move generators are important concepts in this research area, and this talk presented a couple of Clojure libraries for these. With them, you can use STRIPS (Stanford Research Institute Problem Solver) style operators to specify rules and matcher functions to apply them to a set of data.
+
+
+![Pattern matching using matcher library](/img/euroclojure2017/symbolic-ai.png)
+
+### Dependency Heaven, Alex Miller
+
+Alex Miller’s talk was about the obstacles on getting Clojure 1.9 out the door and improving first time use of Clojure for beginners. He started with a recap on Rich Hickey’s keynote from last years Clojure/conj; one point that appeared in conversations to follow was that giving new names to avoid breakage is an underused solution. Also while maven itself is a great service, Clojure doesn’t really care about bytecode artifacts, since it ships with a compiler.
+
+With the separation of Clojure core into multiple artifacts, meaning you can’t run Clojure with a simple “java -cp clojure.jar clojure.main” command anymore, there’s an avenue to rework dependency resolution :). At the same time Clojure is getting an installer, a library for dependency resolution (tools.deps.alpha https://github.com/clojure/tools.deps.alpha) and a startup script (so you can ditch ~/bin/clj :)) which is to be published with brew, apt etc..
+
+
+Tools.deps.alpha resolves dependencies from a deps.edn file, which contains a map, not a vector of dependencies. This means that ordering of dependencies for resolving what version of a dependency to choose is now over and tools.deps.alpha takes care of this. The clj script then reads deps.edn and so could existing tools too. It remains to be seen how this will take on in tooling, but there definitely is a place for fresh take on dependency management here.
+
+![dependency resolution](/img/euroclojure2017/deps.png)
+
+
+### Save the Princess, Simon Belak
+
+Simon Belak’s talk showcased an example on how to build an AI for a simple “game” where you have to find the shortest path to the princess (that you need to save) from a random starting location in an XY-space. The approach was again slightly different from “the usual” back propagating deep neural network trained using reinforcement learning. This time, the AI used evolution strategy (https://blog.openai.com/evolution-strategies/) which seems to be highly parallelizable. It was first trained with a policy network using Cortex, then modelled using Neanderthal and finally the model was also parallelized using Onyx.
```

Comment:
"the usual" viitannee tässä googlen game ai:hin? https://www.theverge.com/2015/2/25/8108399/google-ai-deepmind-video-games